### DISCUSSION TITLE
Can you guys add this feature?

### DISCUSSION THREAD

**Discussion by VictoorH** [File-New-Project/EarTrumpet#943] [Ideas]
Duplicate devices. In the old abandoned Audio Router project you couldn't just route the devices BUT duplicate them too... so you can listen to audio on two or more devices at the same time.
For example, a song playing on Chrome... if I route it to another device any audio playing on Chrome will be muted on my headset. So first I duplicate Chrome in two and one of duplicate Chrome I route it to another device while I can still to listen on my Headset.

Here is a print of this old project that is now dead and unsupported for new Windows updates: https://i.imgur.com/u3PLZTM.png

EarTrumpet its the best alternative for Audio Router that I found, its easy to use, but if it could Duplicate devices it would be even better.

**Comment by Mike-Correa** [comment]
Being able to split the audio from each application into multiple output devices and control each volume separately. That would honestly make EarTrumpet the best mixer software out there. Please make it happen!

**Comment by Earl-Brown** [comment]
Yes!  This would make ET my go-to mixer (especially since windows 10 only has the volume controls in a settings window instead of any reasonable popup UI!)

I am using the defunct audio router mentioned by OP and ... it's clunky.

I even started research in how to do it myself, and it looks like it is quite doable - but I don't know how easy it would be to integrate with existing ET code.

The key tech I would have used is the Windows Audio Graph API.  It's windows 10+ only, so if you wanted backward compatibility, you could use Jack: https://jackaudio.org/api/

Some links for Windows Audio Graph:

https://docs.microsoft.com/en-us/samples/microsoft/windows-universal-samples/audiocreation/` (I think the "Submix Audio Nodes" example is probably the closest to what we'd need here)

This page goes into more detail, in particular stating that "you can create more than one connection from an input node to other nodes" https://docs.microsoft.com/en-us/windows/uwp/audio-video-camera/audio-graphs#node-connections-and-submix-nodes

(In fact, looking at that last one, I can "see" the pieces needed and almost see the code required in my head ...)

An intro tutorial: https://www.html5rocks.com/en/tutorials/webaudio/intro/

The readme from the repo: https://github.com/MicrosoftDocs/windows-uwp/blob/docs/windows-apps-src/audio-video-camera/audio-graphs.md

(Of note, care does need to be taken to avoid excessive CPU or memory use with WAG)

I really hope we can make this a thing!

**Reply by Earl-Brown** [reply]
PS:  Looks like ET is already a C# app...maybe I'll try to add the feature myself?

My plan would be to allow multi-select in the "route this source to that output" UI.

**Reply by Mike-Correa** [reply]
I'm glad to see someone's willing to work on this feature. Thank you! I hope you find a way to make it. 

> My plan would be to allow multi-select in the "route this source to that output" UI.

That would be ideal.
